### PR TITLE
Fixed #23891 -- Moved deprecation of IPAddressField to system check framework.

### PR DESCRIPTION
**Pull request by timgraham**

### CODE REVIEW

**Comment by timgraham** [comment]
Markus, updated #23891 per your comments and added a sketch for how #23861 might look. Will merge the former and continue work on the latter if all looks okay..

File: django/db/models/fields/__init__.py [hunk 1]
```diff
@@ -292,6 +294,30 @@ def _check_null_allowed_for_primary_keys(self):
...
+                    hint=self.removed_for_migrations.get('hint'),
+                    obj=self,
+                    id=self.removed_for_migrations.get('id', 'fields.EXXX'),
+                )
+            ]
+        elif self.deprecated_for_migrations:
+            return [
+                checks.Warning(
+                    self.deprecated_for_migrations.get(
+                        'message', '%s has been deprecated.' % self.__class__.__name__
```
Comment (MarkusH) [inline django/db/models/fields/__init__.py]: Could you wrap this line after the comma.

File: docs/ref/checks.txt [hunk 2]
```diff
@@ -95,6 +95,8 @@ Fields
 * **fields.E160**: The options ``auto_now``, ``auto_now_add``, and ``default``
   are mutually exclusive. Only one of these options may be present.
 * **fields.W161**: Fixed default value provided.
+* **fields.W900**: ``IPAddressField`` has been deprecated. Support for it
+(except in historical migrations) will be removed in Django 1.9.
```
Comment (MarkusH) [inline docs/ref/checks.txt]: Two missing leading spaces.

**Comment by MarkusH** [comment]
Looks good to me, Tim.

**Pull request merged**